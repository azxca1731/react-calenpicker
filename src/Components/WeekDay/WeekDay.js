import React from "react";
import PropTypes from "prop-types";
import { CssConnector } from "../../Containers/Provider";
import style from "./WeekDay.style.scss";

class WeekDay extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { cssObject } = this.props;
    return (
      <thead className={style.WeekDay} style={cssObject}>
        <tr className={style.WeekDay__tr}>
          <td className={style.WeekDay__day}>Sun</td>
          <td className={style.WeekDay__day}>Mon</td>
          <td className={style.WeekDay__day}>Tue</td>
          <td className={style.WeekDay__day}>Wed</td>
          <td className={style.WeekDay__day}>Thu</td>
          <td className={style.WeekDay__day}>Fri</td>
          <td className={style.WeekDay__day}>Sat</td>
        </tr>
      </thead>
    );
  }
}

WeekDay.propTypes = {
  cssObject: PropTypes.object
};

export default CssConnector(({ state }) => ({
  cssObject: state.WeekDayCssObject
}))(WeekDay);
