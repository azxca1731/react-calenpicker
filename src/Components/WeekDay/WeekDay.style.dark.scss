@import "../../Styles/theme/dark";
@import "./WeekDay.style";
