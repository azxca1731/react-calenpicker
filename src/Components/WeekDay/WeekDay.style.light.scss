@import "../../Styles/theme/light";
@import "./WeekDay.style";
