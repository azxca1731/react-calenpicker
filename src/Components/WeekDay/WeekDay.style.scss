.WeekDay {
  display: block;
  height: 13.6%;
  width: 100%;

  &__tr {
    display: block;
    width: 100%;
  }

  &__day {
    display: inline-block;
    padding: 1vh 0.3%;
    width: 13.6%;
    text-align: center;
  }
}
