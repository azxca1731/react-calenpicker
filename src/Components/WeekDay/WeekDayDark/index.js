import WeekDayDark from "./WeekDayDark";

export default WeekDayDark;
