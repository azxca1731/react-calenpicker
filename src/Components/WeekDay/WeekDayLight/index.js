import WeekDayLight from "./WeekDayLight";

export default WeekDayLight;
