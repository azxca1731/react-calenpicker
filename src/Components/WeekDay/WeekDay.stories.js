import React from "react";

import { storiesOf } from "@storybook/react";

import WeekDay from "Components/WeekDay";

storiesOf("Components", module).add("WeekDay", () => <WeekDay />);
