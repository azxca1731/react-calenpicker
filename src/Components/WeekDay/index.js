import WeekDay from "./WeekDay";

export default WeekDay;
