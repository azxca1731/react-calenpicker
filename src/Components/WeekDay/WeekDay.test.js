import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import WeekDay from "./index";

describe("WeekDay", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<WeekDay />);
    expect(wrapper).toHaveLength(1);
  });

  it("renders with snapshot", () => {
    const wrapper = shallow(<WeekDay />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("render custom css render well with snapshot", () => {
    const customCssObject = { color: "red" };
    const wrapper = shallow(<WeekDay cssObject={customCssObject} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(toJson(wrapper).props.style).toBe(customCssObject);
  });
});
