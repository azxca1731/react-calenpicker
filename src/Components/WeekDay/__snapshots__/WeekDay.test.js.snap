// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekDay render custom css render well with snapshot 1`] = `
<styled.thead
  style={
    Object {
      "color": "red",
    }
  }
>
  <styled.tr>
    <styled.td>
      Sun
    </styled.td>
    <styled.td>
      Mon
    </styled.td>
    <styled.td>
      Tue
    </styled.td>
    <styled.td>
      Wed
    </styled.td>
    <styled.td>
      Thu
    </styled.td>
    <styled.td>
      Fri
    </styled.td>
    <styled.td>
      Sat
    </styled.td>
  </styled.tr>
</styled.thead>
`;

exports[`WeekDay renders with snapshot 1`] = `
<styled.thead>
  <styled.tr>
    <styled.td>
      Sun
    </styled.td>
    <styled.td>
      Mon
    </styled.td>
    <styled.td>
      Tue
    </styled.td>
    <styled.td>
      Wed
    </styled.td>
    <styled.td>
      Thu
    </styled.td>
    <styled.td>
      Fri
    </styled.td>
    <styled.td>
      Sat
    </styled.td>
  </styled.tr>
</styled.thead>
`;
